navbar_alerts: Use handlebars instead of django HTML template.

Riken-Shah opened the pull request:
This PR replaces the `navbar_alerts.html` template
with `navbar_alerts.hbs` which will dynamically remove/add
alerts.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->
Tested locally by enabling each alert.


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

Riken-Shah commented on static/js/panels.js:
```diff
@@ -101,24 +104,101 @@ export function show_profile_incomplete(is_profile_incomplete) {
 
+function show_notification_alert() {
+    open({
+        data_process: $t({defaultMessage: "notifications"}),
```
I have created a separate function for this as we are using it twice.

timabbott commented on static/js/panels.js:
```diff
@@ -101,24 +104,101 @@ export function show_profile_incomplete(is_profile_incomplete) {
+function show_notification_alert() {
+    open({
+        data_process: $t({defaultMessage: "notifications"}),
+        first_step_message: $t({defaultMessage: "Zulip needs your permission to"}),
```
You can't split up string this way as different languages have different word orders.  You need instead "Zulip needs your permission to __x__" if you're going to do this, but it'd be better to just have one string.

 https://zulip.readthedocs.io/en/latest/translating/internationalization.html

Riken-Shah replied:
Ohh! Thanks, the docs were very helpful!

timabbott commented on static/js/panels.js:
```diff
@@ -101,24 +104,101 @@ export function show_profile_incomplete(is_profile_incomplete) {
+                    text: $t({defaultMessage: "No, I'll catch up"}),
+                },
+            ],
+        });
```
This is a ton of configuration and feels very hard to read/review; can you think about other ways to do the interface between this and the template?

Riken-Shah commented:
> This is a ton of configuration and feels very hard to read/review; can you think about other ways to do the interface between this and the template?

Yes, @timabbott I agree! 
- I fixed the translation issue in this push by creating sub-templates(partials) for some alerts that need a button/variable between the text block.
- I removed unnecessary translation of the class name and other HTML attributes.

> each panel having a small inner template like navbar_alert_content_bankruptcy

I feel, if we do this, we will just repeat the same code again and again -- we will lose the essence of customization.

Though, We can definitely do this if you didn't like the current idea. 

P.S: If you like this idea I will break down the commit for better readability.

Riken-Shah commented on static/js/navbar_alerts.js:
```diff
@@ -101,24 +104,93 @@ export function show_profile_incomplete(is_profile_incomplete) {
+            data_process: "bankruptcy",
+            class_name: "bankruptcy",
+            is_bankruptcy_alert: true,
+            unread_msgs_count: page_params.unread_msgs.count,
```
Should we also mark `number` for translation?

Riken-Shah commented on static/templates/navbar_alert.hbs:
```diff
@@ -0,0 +1,50 @@
+    <div data-step="1">
+        {{#if is_notification_alert}}
+        {{> notification_alert_first_step_content}}
+        {{else if is_bankruptcy_alert}}
```
My plan here was just to use the `first_step_alert_content` variable (instead of using `is_notification_alert` and `is_bankruptcy_alert`) which will lead to a handlebar file that has the content.

But I couldn't find any relevant helper for it, then I thought I'd create a manual helper but for it, but I can't as we need to read file content and the `fs` module is not allowed on the client-side.

timabbott commented:
Thinking more about this architecture, I think what bugs me is the `second_step` / `first_step` model, and how it leads to a template with a ton of conditionals in it that is hard to read/understand.  I don't understand why we don't just handle the second step as a "second step" mini-template that is parallel to the first.
